client/src/components/Main/index.jsx

```diff
@@ -1,26 +1,93 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 import Modal from '../Modal';
 
 function Main(props) {
+  const [formData, setFormData] = useState({
+    title: '',
+    body: '',
+    status: 'todo',
+  });
+  const [message, setMessage] = useState(null);
+
   const createTask = async () => {
     // TODO: Perform fetch to server providing task details
+
+    const response = await fetch('http://localhost:3000/api/v1/tasks', {
+      method: 'POST',
+      body: JSON.stringify(formData),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
```

Comment:
Here you are missing the Basic Authentication header, you must provide a `Authorization` header with the Basic Authentication approach. As of now we don't support real authentication, instead we provide the User ID in the database as the `username` and the Username (case sensitive) as `password`.

```suggestion
    const response = await fetch('http://localhost:3000/api/v1/tasks', {
      method: 'POST',
      body: JSON.stringify(formData),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Basic ${btoa(user.id + ':' + user.username)}`,
      },
    });
```

You can provide user details as props to `<Main />`.